### DISCUSSION TITLE
Clicking on "attach files" gives Unexpected Application Error!

### DISCUSSION THREAD

**Discussion by wmsanchez** [danny-avila/LibreChat#3649] [Troubleshooting]
### What happened?

After updating to 0.7.4, as soon as you click on the "attach files" button, the application breaks and shows a log of the error.

### Steps to Reproduce

1. Open the app
2. Click on "attach files"

### What browsers are you seeing the problem on?

Chrome, Safari

### Relevant log output

```shell
Unexpected Application Error!
Cannot read properties of null (reading 'includes')
TypeError: Cannot read properties of null (reading 'includes')
    at http://localhost:3080/assets/index-CNbk6mhB.js:3:3779
    at Array.find (<anonymous>)
    at Gr (http://localhost:3080/assets/index-CNbk6mhB.js:3:3769)
    at b4 (http://localhost:3080/assets/index-CNbk6mhB.js:72:18737)
    at s6 (http://localhost:3080/assets/vendor-CSlTTJor.js:38:19539)
    at FO (http://localhost:3080/assets/vendor-CSlTTJor.js:40:44058)
    at DO (http://localhost:3080/assets/vendor-CSlTTJor.js:40:39790)
    at yX (http://localhost:3080/assets/vendor-CSlTTJor.js:40:39718)
    at Kg (http://localhost:3080/assets/vendor-CSlTTJor.js:40:39570)
    at t3 (http://localhost:3080/assets/vendor-CSlTTJor.js:40:35934)
```

### Screenshots

<img width="705" alt="Screenshot 2024-08-14 at 5 28 33 PM" src="https://github.com/user-attachments/assets/82276468-2723-4a08-9ba9-6b9a89e2200b">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can't reproduce this on the latest commit nor on the v0.7.4 tag. Can you give more details? What endpoint do you have selected when you experience this? You should also try it on a private window

**Reply by danny-avila** [reply]
Also more details like your browser, etc. Also please confirm you have updated according to these instructions: https://www.librechat.ai/docs/local/docker#update-librechat

**Reply by ByungsunAhn** [reply]
I get the exact same error.
It is reproduced in 0.7.5 with HuggingFace Clone Deploy and happens when I open the file tab. The files were linked to MongoDB, and I couldn't delete them even if I pressed the delete button. But after updating to 0.7.5, I get this error when I enter the file tab. For reference, the files are still there in MongoDB.

**Reply by mansiibm** [reply]
Hi I am facing this issue with 0.7.6  , for all the endpoints- azureOpenAI, OpenAI,plugins,azureAssistants,Anthropic -  on edge, chrome and incognito. I have tried , cleaning the cache, restart the application several times, but just as soon as I hit to attach files it crashes.  I am using the docker image- ghcr.io/danny-avila/librechat-api:v0.7.6, and I can just use this image as per my code-setup, cannot update this unless a new version is released. 
And this happened for 0.7.5 and 07.6 -both, including their RCs version. 

<img width="960" alt="crashing" src="https://github.com/user-attachments/assets/dd5929d2-8fd2-467f-882d-0663d5932b40" />
Is there any specific reason for this?

**Reply by mansiibm** [reply]
Hi I am experiencing the same issue with the latest build, sharing log for reference- 

Slack trace- 

TypeError: Cannot read properties of null (reading 'startsWith')
   at v7 (https://domain/assets/index.CDTwdDrx.js:3424:19247)
   at h7 (https://domain/assets/vendor.B44UzOen.js:38:19576)
   at Zte (https://domain/assets/vendor.B44UzOen.js:40:44191)
   at Wte (https://domain/assets/vendor.B44UzOen.js:40:39894)
   at a_e (https://domain/assets/vendor.B44UzOen.js:40:39817)
   at U2 (https://domain/assets/vendor.B44UzOen.js:40:39666)
   at m$ (https://domain/assets/vendor.B44UzOen.js:40:36022)
   at UB (https://domain/assets/vendor.B44UzOen.js:40:36827)
   at vd (https://domain/assets/vendor.B44UzOen.js:38:3292)
   at https://domain/assets/vendor.B44UzOen.js:40:34339
[error-log-2025-02-07T11_40_42.177Z.json](https://github.com/user-attachments/files/18705863/error-log-2025-02-07T11_40_42.177Z.json)

**Reply by mansiibm** [reply]
![image](https://github.com/user-attachments/assets/ea3ebc1b-8f64-4c8c-bf16-c36a807ad413)

**Reply by mansiibm** [reply]
Happens when I click on attach files

**Reply by danny-avila** [reply]
@mansiibm can't recreate it, can you double check the files network request is coming through with data?

![chrome_Yza7baEpQM](https://github.com/user-attachments/assets/7d7ec9f9-322f-40e4-b5a6-8633faca9ce9)

make sure to reload the page with the network tab open in chrome inspect

**Reply by ppatayane** [reply]
Hi @danny-avila  Since I am following this, I can share from my side what you asked, if that is helpful.
When **I click on 'My Files' or 'Attach files'** there is nothing that is passed in network. It is empty unlike other navigation, where I can see data processed through network normally. I see same behaviour for Edge, Chrome browsers..

![image](https://github.com/user-attachments/assets/a51073cb-c3a2-4078-a047-bf49a5618b59)


Below is what I see in console..
vendor.B44UzOen.js:40 TypeError: Cannot read properties of null (reading 'startsWith')
    at cell (index.CDTwdDrx.js:3436:41478)
    at h7 (vendor.B44UzOen.js:38:19576)
    at Zte (vendor.B44UzOen.js:40:44191)
    at Wte (vendor.B44UzOen.js:40:39894)
    at a_e (vendor.B44UzOen.js:40:39817)
    at U2 (vendor.B44UzOen.js:40:39666)
    at m$ (vendor.B44UzOen.js:40:36022)
    at UB (vendor.B44UzOen.js:40:36827)
    at vd (vendor.B44UzOen.js:38:3292)
    at vendor.B44UzOen.js:40:34339
r$ @ vendor.B44UzOen.js:40
Mte.o.componentDidCatch.n.callback @ vendor.B44UzOen.js:40
SB @ vendor.B44UzOen.js:38
FB @ vendor.B44UzOen.js:40
zte @ vendor.B44UzOen.js:40
r_e @ vendor.B44UzOen.js:40
l_e @ vendor.B44UzOen.js:40
Ud @ vendor.B44UzOen.js:40
UB @ vendor.B44UzOen.js:40
vd @ vendor.B44UzOen.js:38
(anonymous) @ vendor.B44UzOen.js:40Understand this errorAI
vendor.B44UzOen.js:66 React Router caught the following error during render TypeError: Cannot read properties of null (reading 'startsWith')
    at cell (index.CDTwdDrx.js:3436:41478)
    at h7 (vendor.B44UzOen.js:38:19576)
    at Zte (vendor.B44UzOen.js:40:44191)
    at Wte (vendor.B44UzOen.js:40:39894)
    at a_e (vendor.B44UzOen.js:40:39817)
    at U2 (vendor.B44UzOen.js:40:39666)
    at m$ (vendor.B44UzOen.js:40:36022)
    at UB (vendor.B44UzOen.js:40:36827)
    at vd (vendor.B44UzOen.js:38:3292)
    at vendor.B44UzOen.js:40:34339 {componentStack: '\n    at cell (https://domain/assets/index.CDTwdDrx.js:3069:23209)'}
componentDidCatch @ vendor.B44UzOen.js:66
Mte.o.componentDidCatch.n.callback @ vendor.B44UzOen.js:40
SB @ vendor.B44UzOen.js:38
FB @ vendor.B44UzOen.js:40
zte @ vendor.B44UzOen.js:40
r_e @ vendor.B44UzOen.js:40
l_e @ vendor.B44UzOen.js:40
Ud @ vendor.B44UzOen.js:40
UB @ vendor.B44UzOen.js:40
vd @ vendor.B44UzOen.js:38
(anonymous) @ vendor.B44UzOen.js:40Understand this errorAI

Thank you!

**Reply by mansiibm** [reply]
Hi Danny, thanks for replying. I am sharing this image with you , when I click on files, then it crashes with the following error message
Thankyou.
<img width="943" alt="image" src="https://github.com/user-attachments/assets/8b790e82-c651-411f-8064-f854d577ca4d" />
![image](https://github.com/user-attachments/assets/745982b2-a44d-4c3e-bffa-d4f158a684b7)
So,files are there but Lc yields the error. 


![image](https://github.com/user-attachments/assets/d1d917ad-d3bf-41c4-8e7c-d88c746303a8)

**Reply by danny-avila** [reply]
If the issue can be reproduced in a dev environment, it would be really helpful to follow these instructions to know exactly the point of failure.

https://github.com/danny-avila/LibreChat/discussions/5251#discussioncomment-11800847

The main would be to use the same build and credentials (for mongodb, in case it is related to the file data)

@ppatayane

**Reply by mansiibm** [reply]
Hi Danny thank you for replying, I am using azure container for librechat api-used- 
npm ci
npm  run frontend:dev 
got the following message
![image](https://github.com/user-attachments/assets/fe3723ba-bafe-450b-94cb-6f52143c058b)

Please correct me , if I am wrong, thankyou

**Reply by ppatayane** [reply]
thanks @mansiibm @danny-avila 🤞

**Reply by danny-avila** [reply]
You should run it on a local machine, and not in azure container apps.

**Reply by mansiibm** [reply]
Hi Danny works absolutely fine in local, only error in the container app. 

https://github.com/user-attachments/assets/07905263-2604-4392-9de7-b3a6cc815201

**Reply by mansiibm** [reply]
And when the container app crashes I do not get any logs in the backend just the once I have already shared with you . Thankyou for looking into this today

**Reply by mansiibm** [reply]
Hi Danny, awesome release and great going of LibreChat!! I too used the latest release tag- v7.7-rc1 and my files are still crashing in the container app in azure- 

![image](https://github.com/user-attachments/assets/514e96ab-ba75-4e50-a192-29a356c613ab)

https://github.com/user-attachments/assets/a2d56358-e7a7-4eb8-a179-1b9340932304

Using one replica only, in the backend there are no logs for this. 
But in the UI- 
![image](https://github.com/user-attachments/assets/c1960556-3f8b-450c-9410-f9d2cafe4294)

![image](https://github.com/user-attachments/assets/b0440fe1-b674-4238-94c6-2a666ad2d2c2)

Then I reload the screen and get a stable UI to work with. And my files in the network appear but then again when click on files it crashes. 
 
![image](https://github.com/user-attachments/assets/1f1c9602-648e-4c91-b142-050fdf179082)

This is not working on the edge and chrome I am using incognito to get the above pictures.

**Comment by ppatayane** [comment]
Wanted to share a theory, not sure if relevant for this but worth mentioning. I recently(few weeks) started facing this issue.

1. I had access to shared agent (created by my co-worker). 
2. I could edit the agent, also I tried to delete the file attached. 2 of them together that were there. Save
3. The file were still there after a quick refresh
4. May be in such scenario, file deletion fails in FE but something gets deleted from backend.
5. If we try to access 'My files'/'Attach files' something is out of sync because of above?
